src/window.cpp

```diff
@@ -62,5 +77,62 @@ GLFWwindow* Window::create_window(int width, int height, const char *window_titl
     // Get the width and height of the framebuffer to properly resize the window
     glfwGetFramebufferSize(window, &width, &height);
 
+
+	//glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_HIDDEN); // Don't show cursor
+
     return window;
 }
+
+void Window::error_callback(int error, const char* description)
+{
+	fputs(description, stderr);
+}
+
+void Window::resize_callback(GLFWwindow* window, int width, int height)
+{
+	// Set the viewport size. This is the only matrix that OpenGL maintains for us in modern OpenGL!
+	glViewport(0, 0, width, height);
+
+	float far_plane, fov;
+	Config::config->get_value(Config::str_far_plane, far_plane);
+	Config::config->get_value(Config::str_fov, fov);
+
+	if (height > 0)
+	{
+		for (Scene * s : Aldente::aldente->get_scenes())
+			s->camera->P = glm::perspective(fov, (float)width / (float)height, 0.1f, far_plane);
+	}
+}
+
+void Window::poll_events()
+{
+	glfwPollEvents();
+}
+
+int Window::should_close(GLFWwindow *window)
```

Comment:
What's the point of a class if you just keep passing in pointers? Make window a private variable and use that...